modin/backends/pandas/query_compiler.py

```diff
@@ -202,7 +202,7 @@ def default_to_pandas(self, pandas_op, *args, **kwargs):
             return result
 
     def finalize(self):
-        self._modin_frame.finalize()
+        self._modin_frame._finalize()
```

Comment:
tsk, tsk, using private API here... maybe `BasePandasFrame.finalize()` should be public while query compiler `finalize` should be private?

Reply:
Looking at how we use it (we `finalize` query compiler before pickling it) I suggest declaring `.finalize()` to be public for _both_ query compiler and Modin frame.

Reply:
See the discussion above.